### PR TITLE
PE 426 427 Get assignments data 

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description 
Implementing _get_number_of_graded_assessment and _get_timeliness_of_submissions methods

### CODE REVIEW

File: openedx_proversity_reports/reports/learning_tracker_report.py [hunk 1]
```diff
@@ -31,6 +39,16 @@ def __init__(self, course_id):
...
 
+    @cached_property
+    def assignments_data(self):
+        """
+        Cached property that returns the block structure.
+        """
+        blocks = get_course_in_cache(self.course_key)
+        assignments_data = get_grading_context(get_course_by_id(self.course_key), blocks)
+
+        return assignments_data.get('all_graded_subsections_by_type', {})
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: imo we shouldn't create a constant for every single string, maybe we can follow the suggested rules in this site http://mauveweb.co.uk/posts/2014/12/dont-name-strings.html

[the remark this answers is not in the document]
Comment (Squirrel18) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: Ok, just keep in mind for the next PR reviews... @andrey-canon

File: openedx_proversity_reports/edxapp_wrapper/get_courseware_library.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+""" Backend abstraction """
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply openedx_proversity_reports/edxapp_wrapper/get_courseware_library.py]: added

File: openedx_proversity_reports/reports/learning_tracker_report.py [hunk 3]
```diff
@@ -84,6 +104,31 @@ def _get_cumulative_grade(self, user):
...
+        Number of graded assignment submissions.
+        Args:
+            user: User Model.
+        Returns:
+            Int (Number of graded subsection that has an associated assignment type).
+        """
+        course_grade = get_course_grade_factory().read(user=user, course_key=self.course_key)
+        count = 0
+
+        for assignment_name, subsections_info in six.iteritems(self.assignments_data):
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: changed

File: openedx_proversity_reports/reports/learning_tracker_report.py [hunk 4]
```diff
@@ -94,6 +139,33 @@ def _get_time_bewteen_sessions(self, user):
...
+        Args:
+            user: User Model.
+        Returns:
+            Int (Number of days).
+        """
+        course_grade = get_course_grade_factory().read(user=user, course_key=self.course_key)
+
+        submissions_timeliness = timedelta()
+
+        for assignment_name, subsections_info in six.iteritems(self.assignments_data):
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: changed

File: openedx_proversity_reports/reports/learning_tracker_report.py [hunk 5]
```diff
@@ -84,6 +104,31 @@ def _get_cumulative_grade(self, user):
...
+            user: User Model.
+        Returns:
+            Int (Number of graded subsection that has an associated assignment type).
+        """
+        course_grade = get_course_grade_factory().read(user=user, course_key=self.course_key)
+        count = 0
+
+        for assignment_name, subsections_info in six.iteritems(self.assignments_data):
+            for subsection_info in subsections_info:
+                subsection = subsection_info.get('subsection_block')
```
[the remark this answers is not in the document]
Comment (Squirrel18) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: Address this, you are using this twice. @andrey-canon according to your sugesstion one rule of thumb is if you are using it more than once.

File: openedx_proversity_reports/reports/learning_tracker_report.py [hunk 6]
```diff
@@ -94,6 +139,33 @@ def _get_time_bewteen_sessions(self, user):
...
+        Returns:
+            Int (Number of days).
+        """
+        course_grade = get_course_grade_factory().read(user=user, course_key=self.course_key)
+
+        submissions_timeliness = timedelta()
+
+        for assignment_name, subsections_info in six.iteritems(self.assignments_data):
+            for subsection_info in subsections_info:
+                subsection = subsection_info.get('subsection_block')
```
[the remark this answers is not in the document]
Comment (Squirrel18) [reply openedx_proversity_reports/reports/learning_tracker_report.py]: Address this, you are using this twice. @andrey-canon according to your sugesstion, one rule of thumb is if you are using it more than once.

**Comment by Squirrel18** [comment]
Please squash and merge. @andrey-canon

**Pull request merged**